pkg/storage/tsdb/bucketindex/updater.go

```diff
@@ -223,3 +249,113 @@ func (w *Updater) updateBlockDeletionMarkIndexEntry(ctx context.Context, id ulid
 
 	return BlockDeletionMarkFromThanosMarker(&m), nil
 }
+
+func (w *Updater) updateSeriesDeletionTombstones(ctx context.Context, oldTombstones []*cortex_tsdb.Tombstone) ([]*cortex_tsdb.Tombstone, error) {
+	out := make([]*cortex_tsdb.Tombstone, 0, len(oldTombstones))
+	discovered := make(map[string]cortex_tsdb.BlockDeleteRequestState)
+
+	err := w.bkt.Iter(ctx, "tombstones/", func(s string) error {
+		tName := filepath.Base(s)
+		requestID, state, err := cortex_tsdb.ParseTombstonePath(tName)
+		if err != nil {
+			return err
+		}
+
+		if prevState, exists := discovered[requestID]; !exists {
```

Comment:
I think I've seen this logic or a similar logic in the Tombstone APIs PR. Isn't it better to refactor this?
My recommendation is to move most of this stuff into the tombstones file and just call it from updater.go.

Reply:
Thanks for the suggestion. I refactored it and all the tombstone loading logic is in the tombstones.go file.